tls/s2n_server_new_session_ticket.c

```diff
@@ -313,6 +336,16 @@ S2N_RESULT s2n_tls13_server_nst_write(struct s2n_connection *conn, struct s2n_st
     RESULT_GUARD(s2n_resume_encrypt_session_ticket(conn, output));
     RESULT_GUARD_POSIX(s2n_stuffer_write_vector_size(&ticket_size));
 
+    /* Come back to the ticket lifetime field and write the data */
+    uint32_t ticket_lifetime_in_secs = 0;
+    RESULT_GUARD(s2n_generate_ticket_lifetime(conn, conn->ticket_fields.key_intro_time, conn->ticket_fields.current_time, &ticket_lifetime_in_secs));
```

Comment:
In this TLS1.3 version, s2n_generate_ticket_lifetime was always GUARD'd. Was that safe? What's the difference between how failures are handled in the TLS1.2 version of this method vs the TLS1.3 version?